### PR TITLE
Misc tweaks

### PR DESCRIPTION
**Pull request by aidanhs**
```
    pub fn finish(&mut self) -> Vec<Edge> {
        // What is this meant to be doing?
        if self.bytes_chunk != 0 || self.bytes_total == 0 {
            self.edge_found(0);
        }
        mem::replace(&mut self.edges, vec!())
    }
```

What is this meant to be doing? It looks a bit strange.

(not ready to merge)

### CODE REVIEW

**Comment by dpc** [comment]
At the end of each data to be chunked, an edge needs to be generated. The condition is: either there was something left from the last edge, or the whole file was zero bytes.

File: Cargo.toml [hunk 1]
```diff
@@ -9,14 +9,14 @@ documentation = "https://dpc.github.io/rdedup-lib/"
...
-
+exclude = ["lib/**/*"]
 
 [[bin]]
 name = "rdedup"
 path = "src/bin.rs"
 
 [dependencies]
-rdedup-lib = "0.0.1"
+rdedup-lib = { version = "0.0.1", path = "lib" }
```
Comment (dpc) [inline Cargo.toml]: This works with both crates.io and local git? Nice. I was wondering if I can get it working like that.

File: lib/src/lib.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ impl Chunker {
         self.edges.push((input_ofs, sha256));
 
         self.chunks_total += 1;
-        self.bytes_chunk += 0;
+        self.bytes_chunk = 0;
```
Comment (dpc) [inline lib/src/lib.rs]: Wow. Good catch.

**Comment by dpc** [comment]
Thanks!

**Comment by aidanhs** [comment]
> At the end of each data to be chunked, an edge needs to be generated. The condition is: either there was something left from the last edge, or the whole file was zero bytes.

Ok, but that's not what doing I think? It's generating an edge at the beginning of the chunk, per `self.edge_found(0);`. I think `0` needs replacing with `self.bytes_total`.

**Comment by dpc** [comment]
I see what you mean. The way code works is confusing. Each `input(buf : &[u8])` returns `Vec<usize>` which contains offsets where the edges were found. Offsets inside the `buf`, which is different then offsets from the last edge.

`finish()` is kind of a `input(&vec!())`, which means if there's and edge it is found right after the end of argument buffer, which is offset 0.

**Pull request merged**